README
Dist::Zilla::PluginBundle::Author::KENTNL

This is the plug-in bundle that KENTNL uses. It exists mostly because he
is very lazy and wants others to be using what he's using if they want
to be doing work on his modules.

INSTALLATION

This is a Perl module distribution. It should be installed with whichever
tool you use to manage your installation of Perl, e.g. any of

  cpanm .
  cpan  .
  cpanp -i .

Consult http://www.cpan.org/modules/INSTALL.html for further instruction.
Should you wish to install this module manually, the procedure is

  perl Makefile.PL
  make
  make test
  make install

COPYRIGHT AND LICENSE

This software is copyright (c) 2016 by Kent Fredric
<kentfredric@gmail.com>.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


Changes
Revision history for Dist-Zilla-PluginBundle-Author-KENTNL

2.025022 2017-03-04T03:21:07Z d50c8ff

2.025021 2017-03-04T02:54:50Z 57634ed
 [Bugfix]
 - Avoid test failures on 5.26 w/ -Ddefault_inc_excludes_dot
 - Closes https://github.com/kentnl/Dist-Zilla-PluginBundle-Author-KENTNL/issues/3

 [Dependencies::Stats]
 - Dependencies changed since 2.025020, see misc/*.deps* for details
 - develop: +1 ↑1
 - runtime: ↑1 -1 (recommends: -5)
 - test: ↓1

 [Internals]
 - Remove usage of ConfigDumper

2.025020 2016-06-19T07:19:45Z 9153e61
 [Dependencies::Stats]
 - Dependencies changed since 2.025010, see misc/*.deps* for details
 - test: ↑1 -1

 [Features]
 - 'bumpversions' moved to 'bump_versions', but legacy interface retained.
 - 'srcreadme' moved to 'src_readme' but legacy interface retained.
 - 'copyfiles' moved to 'copy_files' but legacy interface retained.
 - Remove residual reliance on Path::Class ( Requires DZil >5.038 )( Closes #1 )

2.025010 2015-11-15T02:28:17Z f61f784
 - Provision more files during minting instead of relying on generation on the first invocation of `dzil build`

2.025003 2015-10-19T14:11:40Z 834555c
 [Bugfix]
 - Avoid test failures on Dzil 5.040 by not expecting pod-* tests anymore.

 [Dependencies::Stats]
 - Dependencies changed since 2.025002, see misc/*.deps* for details
 - develop: +2 ↑1

2.025002 2015-08-14T12:13:53Z 6d20907
 [Bugfix]
 - Ensure declared minimum version of MinimumPerl
 - Provision a bunch of stock files I tend to need.

 [Dependencies::Stats]
 - Dependencies changed since 2.025001, see misc/*.deps* for details
 - configure: ↓1
 - runtime: ↑1 ↓1 -1
 - test: ↓1

2.025001 2015-07-05T03:52:11Z 31e79e3
 [Bugfix]
 - Use a newer contributing module in pre-emption of ether accepting my patch

 [Dependencies::Stats]
 - Dependencies changed since 2.025000, see misc/*.deps* for details
 - develop: ↑1
 - runtime: ↑1

 [Note]
 - I really hate having so many plugins, and I really hate everyone needing my bundle
 - But I also hate anyone needing my bundle.

 [Todo]
 - Have a system where more static stuff happens at a dedicated phases instead of part of the build cycle
 - Not have contributors need to execute that to simply refresh files that are already there.

2.025000 2015-06-30T02:35:26Z 8f4450c
 [00 Features]
 - copyfiles now copies LICENSE and Makefile.PL by default.
 - RemovePrereqs::Provided is now standard issue.
 - default_branch now specified for headless Git::NextRelease
 - Inject a CONTRIBUTING document standard

 [Dependencies::Stats]
 - Dependencies changed since 2.024002, see misc/*.deps* for details
 - develop: +1 (suggests: -1)
 - runtime: +2

2.024002 2015-06-21T22:23:14Z b786cb3
 [Dependencies::Stats]
 - Dependencies changed since 2.024001, see misc/*.deps* for details
 - develop: +2 -2 (suggests: ↑2)
 - runtime: +1 -1

 [Minor]
 - Now uses MetaYAML::Minimal to generate a reduced META.yml

2.024001 2015-03-01T10:38:03Z 57fafd2
 [Bugfix]
 - Closing the loop with inherently co-depedendent changes.

 [Dependencies::Stats]
 - Dependencies changed since 2.024000, see misc/*.deps* for details
 - develop: +1 ↑2 (suggests: ↑1)
 - runtime: ↑2
 - test: ↑1

2.024000 2015-02-28T17:36:05Z ffe5dea
 [Compatibility Breaking Changes]
 - CommitBuild now builds to 'builds' regardless what branch you're on. This is mostly to negate hassle  involved with
   working on multiple branches blocking activity due to 'dzil build' doing the wrong thing and creating new, disjoint
   branches on its own.

 [Dependencies::Stats]
 - Dependencies changed since 2.023003, see misc/*.deps* for details
 - develop: -1 (suggests: ↑1)

2.023003 2015-02-05T10:55:56Z 66ca3cb
 [Bugfix]
 - Ensure quoted YAML via Travis stack upgrades.

 [Dependencies::Stats]
 - Dependencies changed since 2.023002, see misc/*.deps* for details
 - develop: ↑1 (suggests: ↑1)
 - runtime: (recommends: +1)

2.023002 2015-01-14T07:26:23Z cdffce2
 [Dependencies::Stats]
 - Dependencies changed since 2.023001, see misc/*.deps* for details
 - configure: (recommends: ↓1)
 - develop: +1 ↓56 -1 (recommends: ↓1, suggests: +1)
 - runtime: +1 ↓59 -16 (recommends: +18)
 - test: ↓11 (recommends: +2 ↓3)

 [Features]
 - Use a less aggressive technique for recommending newer EUMM/TB to be slightly more stable and do 99% of what I mean
   without so much commit churn.
 - Selected packages auto-recommend selected versions as implied upgrade ( much more preceise and less automatic than
   all my random 'MatchInstalled' rubbish

 [Packaging]
 - Moved a boatload of hard required deps that were really mostly recommendations to where they belong.
 - Drop "ramp all the things" version insanity ( caveat: more things are going to break now due to not knowing all
   minimum deps )

2.023001 2015-01-13T12:12:42Z 9c13379
 [Bugfix]
 - Stop injecting TravisCI in newly minted dists ( superceded by bundle changes )

 [Dependencies::Stats]
 - Dependencies changed since 2.023000, see misc/*.deps* for details
 - configure: (recommends: ↑1)
 - develop: ↑2
 - runtime: ↑3
 - test: (recommends: ↑1)

2.023000 2015-01-13T08:13:40Z 8d47bcc
 [API]
 - Missing constructor arguments are now only warnings
 - Now does :PluginRemover and :Config::Slicer plugin bundle roles.
 - Inject TravisCI plugin standard.

 [Dependencies::Stats]
 - Dependencies changed since 2.022005, see misc/*.deps* for details
 - develop: ↑18 (recommends: ↑1)
 - runtime: +3 ↑21 -2
 - test: ↑2 (recommends: ↑1)

 [Internals]
 - No more MX::AttributeShortcuts
 - No more StrictConstructor

2.022005 2015-01-05T20:16:46Z cba7b74
 [Dependencies::Stats]
 - Dependencies changed since 2.022004, see misc/*.deps* for details
 - develop: +1 ↑18 -1
 - runtime: +1 ↑21 -1
 - test: ↑2

 [Minor]
 - Switch to Readme::Brief

2.022004 2015-01-02T03:00:30Z 07c5593
 [Dependencies::Stats]
 - Dependencies changed since 2.022003, see misc/*.deps* for details
 - develop: +1 ↑1
 - runtime: ↑4
 - test: +1

 [Minor]
 - use new travis generator
 - inject new travis generator into new dists

Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.008.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;

use File::ShareDir::Install;
$File::ShareDir::Install::INCLUDE_DOTFILES = 1;
$File::ShareDir::Install::INCLUDE_DOTDIRS = 1;
install_share module => "Dist::Zilla::MintingProfile::Author::KENTNL", "share\/profiles";


my %WriteMakefileArgs = (
  "ABSTRACT" => "BeLike::KENTNL when you build your distributions.",
  "AUTHOR" => "Kent Fredric <kentnl\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::ShareDir::Install" => "0.10"
  },
  "DISTNAME" => "Dist-Zilla-PluginBundle-Author-KENTNL",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::PluginBundle::Author::KENTNL",
  "PREREQ_PM" => {
    "Carp" => 0,
    "Dist::Zilla" => "5.011",
    "Dist::Zilla::File::FromCode" => 0,
    "Dist::Zilla::Plugin::Author::KENTNL::CONTRIBUTING" => "0.001003",
    "Dist::Zilla::Plugin::Author::KENTNL::RecommendFixes" => "0.004002",
    "Dist::Zilla::Plugin::Author::KENTNL::TravisCI" => "0.001002",
    "Dist::Zilla::Plugin::Authority" => "1.006",
    "Dist::Zilla::Plugin::AutoPrereqs" => 0,
    "Dist::Zilla::Plugin::BumpVersionAfterRelease" => 0,
    "Dist::Zilla::Plugin::ConfirmRelease" => 0,
    "Dist::Zilla::Plugin::CopyFilesFromBuild" => 0,
    "Dist::Zilla::Plugin::Git::Check" => 0,
    "Dist::Zilla::Plugin::Git::Commit" => 0,
    "Dist::Zilla::Plugin::Git::CommitBuild" => 0,
    "Dist::Zilla::Plugin::Git::Contributors" => "0.006",
    "Dist::Zilla::Plugin::Git::GatherDir" => 0,
    "Dist::Zilla::Plugin::Git::NextRelease" => "0.004000",
    "Dist::Zilla::Plugin::Git::Tag" => 0,
    "Dist::Zilla::Plugin::GithubMeta" => 0,
    "Dist::Zilla::Plugin::License" => 0,
    "Dist::Zilla::Plugin::MakeMaker" => 0,
    "Dist::Zilla::Plugin::Manifest" => 0,
    "Dist::Zilla::Plugin::ManifestSkip" => 0,
    "Dist::Zilla::Plugin::MetaConfig" => 0,
    "Dist::Zilla::Plugin::MetaData::BuiltWith" => "1.004000",
    "Dist::Zilla::Plugin::MetaJSON" => 0,
    "Dist::Zilla::Plugin::MetaProvides::Package" => "1.14000001",
    "Dist::Zilla::Plugin::MetaTests" => 0,
    "Dist::Zilla::Plugin::MetaYAML::Minimal" => 0,
    "Dist::Zilla::Plugin::MinimumPerl" => "1.004",
    "Dist::Zilla::Plugin::PodCoverageTests" => 0,
    "Dist::Zilla::Plugin::PodSyntaxTests" => 0,
    "Dist::Zilla::Plugin::PodWeaver" => 0,
    "Dist::Zilla::Plugin::Prereqs" => 0,
    "Dist::Zilla::Plugin::Prereqs::AuthorDeps" => 0,
    "Dist::Zilla::Plugin::Prereqs::Recommend::MatchInstalled" => 0,
    "Dist::Zilla::Plugin::Prereqs::Upgrade" => 0,
    "Dist::Zilla::Plugin::Readme::Brief" => 0,
    "Dist::Zilla::Plugin::ReadmeAnyFromPod" => 0,
    "Dist::Zilla::Plugin::RemovePrereqs::Provided" => 0,
    "Dist::Zilla::Plugin::RewriteVersion::Sanitized" => 0,
    "Dist::Zilla::Plugin::RunExtraTests" => 0,
    "Dist::Zilla::Plugin::Test::CPAN::Changes" => 0,
    "Dist::Zilla::Plugin::Test::Compile::PerFile" => "0.003902",
    "Dist::Zilla::Plugin::Test::EOL" => 0,
    "Dist::Zilla::Plugin::Test::Kwalitee" => 0,
    "Dist::Zilla::Plugin::Test::MinimumVersion" => 0,
    "Dist::Zilla::Plugin::Test::Perl::Critic" => 0,
    "Dist::Zilla::Plugin::Test::ReportPrereqs" => 0,
    "Dist::Zilla::Plugin::TestRelease" => 0,
    "Dist::Zilla::Plugin::Twitter" => 0,
    "Dist::Zilla::Plugin::UploadToCPAN" => 0,
    "Dist::Zilla::Role::BundleDeps" => 0,
    "Dist::Zilla::Role::FileGatherer" => 0,
    "Dist::Zilla::Role::MintingProfile::ShareDir" => 0,
    "Dist::Zilla::Role::PluginBundle" => 0,
    "Dist::Zilla::Role::PluginBundle::Config::Slicer" => 0,
    "Dist::Zilla::Role::PluginBundle::PluginRemover" => 0,
    "Dist::Zilla::Util::CurrentCmd" => 0,
    "Moose" => 0,
    "Moose::Util::TypeConstraints" => 0,
    "String::Formatter" => 0,
    "namespace::autoclean" => 0,
    "strict" => 0,
    "version" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Capture::Tiny" => 0,
    "Dist::Zilla::Plugin::Author::KENTNL::TravisCI" => "0.001002",
    "Dist::Zilla::Plugin::Git::NextVersion::Sanitized" => 0,
    "Dist::Zilla::Plugin::ModuleBuild" => 0,
    "Dist::Zilla::Plugin::Prereqs::MatchInstalled" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "File::pushd" => 0,
    "FindBin" => 0,
    "Git::Wrapper" => 0,
    "Module::Build" => 0,
    "Module::Metadata" => 0,
    "Path::Tiny" => 0,
    "Test::DZil" => "5.038",
    "Test::Fatal" => 0,
    "Test::File::ShareDir" => "0.3.0",
    "Test::More" => 0,
    "Test::Output" => 0,
    "lib" => 0
  },
  "VERSION" => "2.025022",
  "test" => {
    "TESTS" => "t/*.t t/00-compile/*.t"
  }
);


my %FallbackPrereqs = (
  "Capture::Tiny" => 0,
  "Carp" => 0,
  "Dist::Zilla" => "5.011",
  "Dist::Zilla::File::FromCode" => 0,
  "Dist::Zilla::Plugin::Author::KENTNL::CONTRIBUTING" => "0.001003",
  "Dist::Zilla::Plugin::Author::KENTNL::RecommendFixes" => "0.004002",
  "Dist::Zilla::Plugin::Author::KENTNL::TravisCI" => "0.001002",
  "Dist::Zilla::Plugin::Authority" => "1.006",
  "Dist::Zilla::Plugin::AutoPrereqs" => 0,
  "Dist::Zilla::Plugin::BumpVersionAfterRelease" => 0,
  "Dist::Zilla::Plugin::ConfirmRelease" => 0,
  "Dist::Zilla::Plugin::CopyFilesFromBuild" => 0,
  "Dist::Zilla::Plugin::Git::Check" => 0,
  "Dist::Zilla::Plugin::Git::Commit" => 0,
  "Dist::Zilla::Plugin::Git::CommitBuild" => 0,
  "Dist::Zilla::Plugin::Git::Contributors" => "0.006",
  "Dist::Zilla::Plugin::Git::GatherDir" => 0,
  "Dist::Zilla::Plugin::Git::NextRelease" => "0.004000",
  "Dist::Zilla::Plugin::Git::NextVersion::Sanitized" => 0,
  "Dist::Zilla::Plugin::Git::Tag" => 0,
  "Dist::Zilla::Plugin::GithubMeta" => 0,
  "Dist::Zilla::Plugin::License" => 0,
  "Dist::Zilla::Plugin::MakeMaker" => 0,
  "Dist::Zilla::Plugin::Manifest" => 0,
  "Dist::Zilla::Plugin::ManifestSkip" => 0,
  "Dist::Zilla::Plugin::MetaConfig" => 0,
  "Dist::Zilla::Plugin::MetaData::BuiltWith" => "1.004000",
  "Dist::Zilla::Plugin::MetaJSON" => 0,
  "Dist::Zilla::Plugin::MetaProvides::Package" => "1.14000001",
  "Dist::Zilla::Plugin::MetaTests" => 0,
  "Dist::Zilla::Plugin::MetaYAML::Minimal" => 0,
  "Dist::Zilla::Plugin::MinimumPerl" => "1.004",
  "Dist::Zilla::Plugin::ModuleBuild" => 0,
  "Dist::Zilla::Plugin::PodCoverageTests" => 0,
  "Dist::Zilla::Plugin::PodSyntaxTests" => 0,
  "Dist::Zilla::Plugin::PodWeaver" => 0,
  "Dist::Zilla::Plugin::Prereqs" => 0,
  "Dist::Zilla::Plugin::Prereqs::AuthorDeps" => 0,
  "Dist::Zilla::Plugin::Prereqs::MatchInstalled" => 0,
  "Dist::Zilla::Plugin::Prereqs::Recommend::MatchInstalled" => 0,
  "Dist::Zilla::Plugin::Prereqs::Upgrade" => 0,
  "Dist::Zilla::Plugin::Readme::Brief" => 0,
  "Dist::Zilla::Plugin::ReadmeAnyFromPod" => 0,
  "Dist::Zilla::Plugin::RemovePrereqs::Provided" => 0,
  "Dist::Zilla::Plugin::RewriteVersion::Sanitized" => 0,
  "Dist::Zilla::Plugin::RunExtraTests" => 0,
  "Dist::Zilla::Plugin::Test::CPAN::Changes" => 0,
  "Dist::Zilla::Plugin::Test::Compile::PerFile" => "0.003902",
  "Dist::Zilla::Plugin::Test::EOL" => 0,
  "Dist::Zilla::Plugin::Test::Kwalitee" => 0,
  "Dist::Zilla::Plugin::Test::MinimumVersion" => 0,
  "Dist::Zilla::Plugin::Test::Perl::Critic" => 0,
  "Dist::Zilla::Plugin::Test::ReportPrereqs" => 0,
  "Dist::Zilla::Plugin::TestRelease" => 0,
  "Dist::Zilla::Plugin::Twitter" => 0,
  "Dist::Zilla::Plugin::UploadToCPAN" => 0,
  "Dist::Zilla::Role::BundleDeps" => 0,
  "Dist::Zilla::Role::FileGatherer" => 0,
  "Dist::Zilla::Role::MintingProfile::ShareDir" => 0,
  "Dist::Zilla::Role::PluginBundle" => 0,
  "Dist::Zilla::Role::PluginBundle::Config::Slicer" => 0,
  "Dist::Zilla::Role::PluginBundle::PluginRemover" => 0,
  "Dist::Zilla::Util::CurrentCmd" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "File::pushd" => 0,
  "FindBin" => 0,
  "Git::Wrapper" => 0,
  "Module::Build" => 0,
  "Module::Metadata" => 0,
  "Moose" => 0,
  "Moose::Util::TypeConstraints" => 0,
  "Path::Tiny" => 0,
  "String::Formatter" => 0,
  "Test::DZil" => "5.038",
  "Test::Fatal" => 0,
  "Test::File::ShareDir" => "0.3.0",
  "Test::More" => 0,
  "Test::Output" => 0,
  "lib" => 0,
  "namespace::autoclean" => 0,
  "strict" => 0,
  "version" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);

{
package
MY;
use File::ShareDir::Install qw(postamble);
}
